[5.11.0] JS API reference update (#3571)
wso2/docs-is: en/docs/references/adaptive-authentication-js-api-reference.md

var state = getValueFromDecodedAssertion(context.request.params.request[0],"state",true);
```

#### getUniqueUserWithClaimValues(claimMap, context, profile)

> The utility function will search on the underlying user stores and return a unique user with the claim values. The claim map will consist of the claim and value. The function will get the 1st key from the map and get all users with the claim and add to list. Then remove the users from that list not having the other claims. So the order of the map will decide the performance of this function.

Reviewer: ```suggestion
The utility function will search on the underlying user stores and return a unique user with the claim values. The claim map will consist of the claim and value. The function will get the first key from the map and get all users with the claim and add them to the list. Then, remove the users that do not have the other claims from that list. Thereby, the order of the map will decide the performance of this function.
```

The first claim in this must have a low number of users.

| Parameter            | Description                                                                  |
|----------------------|------------------------------------------------------------------------------|
| claimMap  | A map contains the claim URI and claim value.    |
| context   | The authentication context, which contains the context information about the request.    |

> | parameterName | Profile of the user. (Optional, the default value is 'default')      |

Reviewer: ```suggestion
| parameterName | (Optional) Profile of the user. The default value is 'default').     |
```

``` java
var claimMap = {};
claimMap[MAPPED_FEDERATED_USER_NAME_CLAIM] = federatedUserName;
claimMap[MAPPED_FEDERATED_IDP_NAME_CLAIM] = idpName;
var mappedUsername = getUniqueUserWithClaimValues(claimMap, context);
```

#### getAssociatedLocalUser(federatedUser)

> This function prompts user input. It includes the following parameters.

Reviewer: ```suggestion
This function prompts user input. It includes the following parameters:
```
